### PR TITLE
feat: ✨ add types for node-replicate

### PR DESCRIPTION
**Pull request by Ankan002**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Please consider adding `package.json`:
```json
{
    "private": true,
    "type": "module"
}
```

File: types/node-replicate/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    prompt: string;
+}
+
+export class Prediction {
+    constructor(props?: Record<string, string>);
+    hasTerminalStatus(): boolean;
+    load(): Promise<PredictionResponse>;
+}
+
+export class Model {
```
Comment (peterblazejewicz) [inline types/node-replicate/index.d.ts:62]:
```suggestion
declare class Prediction {
    constructor(props?: Record<string, string>);
    hasTerminalStatus(): boolean;
    load(): Promise<PredictionResponse>;
}

declare class Model {
```
those are not exported from a module directly, if you want to expose them, but without adding an export from their respective modules, you can:
```ts
export default replicate;

export type { Model, Prediction };
```

**Comment by Ankan002** [comment]
@peterblazejewicz I added the package.json with the following properties only, if any more properties are required I can add that. Although a check config flag got added so was worried that there might be configs missing in package.json

**Comment by peterblazejewicz** [comment]
> @peterblazejewicz I added the package.json with the following properties only, if any more properties are required I can add that. Although a check config flag got added so was worried that there might be configs missing in package.json

check config will be ack here, the type module is expected addition

**Comment by Ankan002** [comment]
@peterblazejewicz just updated the module rule. Give it a check if anything more change is required.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Ankan002 thanks!

**Comment by Ankan002** [comment]
Ready to merge

**Comment by Ankan002** [comment]
@peterblazejewicz Thank you!!

**Pull request merged**